### PR TITLE
agent: Unify agent  logic to crate agent-utils

### PR DESCRIPTION
**Pull request by ecpullen**
**Issue number:**
N/A

**Description of changes:**

This change moves commonly used functions of bottlerocket agents to their own crate that can be used by other agents.

**Testing done:**

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

File: bottlerocket/agents/src/error.rs [hunk 1]
```diff
@@ -184,4 +153,8 @@ pub enum Error {
 
     #[snafu(display("The task definition is missing"))]
     TaskDefinitionMissing,
+
+    #[snafu(context(false))]
+    #[snafu(display("{}", source))]
+    Utils { source: agent_utils::Error },
```
Comment (etungsten) [inline bottlerocket/agents/src/error.rs:159]: I can't seem to find any references to this enum variant. Can we remove this?

**Review by etungsten — APPROVED** [review]

File: bottlerocket/agents/src/error.rs (hunk 1 above)
Comment (ecpullen) [reply bottlerocket/agents/src/error.rs:159]: That is a wrapper for any error that happened during a utils call. It is used, but snafu automatically converts it from utils::Error to bottlerocket_agents::Error.

Comment (etungsten) [reply bottlerocket/agents/src/error.rs:159]: I see. `#[snafu(context(false))]` let's us not have to explicitly call `.context(UtilsSnafu)`. That's neat.

**Review by stmcginnis — APPROVED** [review]
lgtm

**Pull request merged**